app/models/ext_management_system.rb

```diff
@@ -673,17 +683,25 @@ def self.destroy_queue(ids)
     find(Array.wrap(ids)).map(&:destroy_queue)
   end
 
-  def destroy_queue
+  def destroy_queue(queue_options = {})
     msg = "Queuing destroy of #{self.class.name} with id: #{id}"
 
     _log.info(msg)
+
+    # Before we queue the `#destroy` pause the provider to prevent a provider in
+    # a bad state from filling up the queue preventing the `#destroy` from being
+    # processed.
+    pause_queue!(:priority => MiqQueue::HIGH_PRIORITY)
+
     task = MiqTask.create(
       :name    => "Destroying #{self.class.name} with id: #{id}",
       :state   => MiqTask::STATE_QUEUED,
       :status  => MiqTask::STATUS_OK,
-      :message => msg,
+      :message => msg
     )
-    self.class._queue_task('destroy', [id], task.id)
+
+    self.class._queue_task('orchestrate_destroy', [id], task.id, queue_options)
```

Comment:
Can we move this queueing into a method called `orchestrate_destroy_queue`? Then this method reads really clean.

Reply:
Done